### ISSUE TITLE
simple permissions, some options are not "moderation"

### ISSUE THREAD

**Issue by illori** [SimpleMachines/SMF#509]
if you look at the simple layout for permissions some options that are under moderate the board are not really moderation tasks.

send topics to friends is one that i see off hand, others are options related to doing things to _your_ post/thread so not really board moderation either.

IMO the simple permissions layout should be removed and only have one to support.

**Comment by illori** [comment]
also

'view_attachments' => array(false, 'attachment', 'participate'),
'post_unapproved_attachments' => array(false, 'attachment', 'make_unapproved_posts'),
'post_attachment' => array(false, 'attachment', 'attach'),

why have the permissions for attachments in several different places in the simple view?

i bet there are more issues with this layout and where permissions are.

not to mention there is a checkbox for each category... it expands/collapses the section and may check/uncheck the options for that section, which is annoying if you uncheck and it collapses the section... so you need to use the array or heading to expand the section to see that the options are what you want

**Comment by Oldiesmann** [comment]
The simple view was added in 2.0 to make permissions a bit more user-friendly and less confusing. We can sort out the permissions issues, but removing that view simply because some permissions aren't in the right areas isn't a good idea.

**Comment by illori** [comment]
no the simple view is not more user friendly, for that matter given the issues i posted above it is less user friend and more code for the developers to keep updated as time goes on. yes we can move things around and sort it better, but it is not any more useful then the old classic permissions that already exists.

**Comment by illori** [comment]
also related to this, why do we have a separate section for post moderation next to edit profiles? i have never used it nor think it is useful. for that matter looking at it, it seems very confusing and hard to understand.

**Comment by NanoSector** [comment]
Anyone who will be messing with permissions has to know that they can screw up things badly, so I don't get the point of the simple view at all. The regular view is pretty user-friendly, too. The simple view only confused me because of missing permissions and a weird layout.

**Comment by Oldiesmann** [comment]
The simple view, at least in 2.1, seems perfectly useable to me. Also, where is post moderation next to edit profiles? They shouldn't be in the same section as edit profiles is a global permission while post moderation is board-specific.

**Comment by illori** [comment]
when you look at the sub menu for permissions, there is a spot for post moderation.

index.php?action=admin;area=permissions;sa=postmod

**Comment by Oldiesmann** [comment]
Oh. That page. I thought you were talking about something different.

That page needs to stay. It's an extremely useful way to set up post moderation options, especially if you're confused by the permissions.

Also, @Yoshi2889 - what permissions do you see missing? Every permission that can be set in the standard ("classic") view should be in the simple view as well.

**Comment by illori** [comment]
oldies most people dont use that page so it does not need to stay. it is very confusing and not helpful in anyway.

i have not ever seen a user on sm.org even ask about it, so that means it is not used. people use permissions not that dumb page, it is useless and a waste of space.

**Comment by NanoSector** [comment]
@Oldiesmann There are so many permissions moved around in that view, it is hard to see if it is not there or if it is in a completely unrelated category, which in turn makes the page very confusing.

**Comment by Oldiesmann** [comment]
illori - What's "very confusing" about that page? You go to that page, select a permissions profile, select the appropriate radio button(s) for each group and click save. If you don't know what each column is for, you simply hover over the icon at the top of each column or look at the legend at the bottom of the page. How much simpler can we make it?

Yoshi - I'll take a look and see if I can find anything that seems out of place. In 2.1 they all seem to be grouped logically, but we can take a thorough look at things before 2.1 goes final and reorganize/change things as needed.

**Comment by ghost** [comment]
For what its worth I say leave it and improve the documentation if its that confusing, saying that if no-one is asking maybe that means they understand, either way no reason to remove.

**Comment by illori** [comment]
oldies that page does not have permission profiles on it, the fact that you need to use a legend to figure out what the columns are talking about does not help make it user friendly at all. i doubt the page is used and just takes up useless space.

i am not sure how to document the page as it is a duplicate of the existing function you can get by setting permissions in the normal area.

**Comment by emanuele45** [comment]
Quick note, it may be already tracked, it may not, I don't remember at the moment.

The post moderation page is broken when it comes to guests (own/any) permissions, I'm sure there is a report in the bug reports board.

Also:
- I personally never used the simple view because I'm used to the "difficult" (even though I find it simpler) one.
- @smfrunic if you have something that should be **simpler** (than something else) but that requires more documentation (than the counterpart) it means it is not simple, but it is at least as complex as the previous one. In general, when creating an application **for users**, documentation should be the last resort for very complicated matters that really require documentation, not for day-to-day things (and from what I understood at simplemachines.org permissions is for most of the admins a daily job, no idea why of course).
  If it is not used it means it is not what the user is expecting and even from a support point of view is easier to say "switch to classic (that is more intuitive...heh)" rather than try to explain the "simple" view. ;)

My 2 cents as usual. :wink:

**Comment by Oldiesmann** [comment]
@illori - It does. It's at the top right of the page in the form of a dropdown list.

**Comment by illori** [comment]
@Oldiesmann that does not make it any easier to use oldies and as emanuele says it has a bug, it is really useless.

**Comment by Oldiesmann** [comment]
We can fix the bug and adjust the layout if needed. One bug does not make an entire feature useless, and I have no plans to remove that page.

**Comment by illori** [comment]
well oldies it is not just adjusting the layout, the page is a complete duplicate of already existing function, no one uses it and no one discusses it, so it is not useful or used. fixing the bug would be nice but IMO a waste of time and the page should just be deleted. if you want to waste peoples time fixing bugs in stuff that is duplicated function elsewhere go ahead but i dont see it being useful time spent.

**Comment by emanuele45** [comment]
> One bug does not make an entire feature useless

Quoting for reference.

**Comment by ghost** [comment]
ok looking better, I think that Simple and Classic should both stay, however the Post Moderation Permissions Page:

http://prntscr.com/1fkzkq

Is quite confusing if you have many groups so I can see the issue with this

**Comment by Arantor** [comment]
I agree that the simple view isn't very simple, and while there is only a small 'bug' if you will, it doesn't make the feature useless in itself, true enough. But the fact that it's not a _good_ feature DOES make the feature less useful to have. There is little value in having two (or for some jobs, three) different interfaces that do the same thing. It isn't simple, it isn't elegant, it isn't efficient. I'd personally not mind seeing it go (which, don't forget, also has knock on consequences for mod authors, for translators and whatnot) but if we're not going to kill it, let's at least solve the issue at hand.

Some of the things are not in the ideal place. Send topics being the first example... it isn't in the right place, it's not a moderation task. More relevantly, though... do we actually need it? Would we not be better served offering up links to the conventional social networks instead? Hell, if we didn't go that far, removing send topic to fiends... it is not a particularly useful feature these days (and it is more problematic than you might think, being both a feature that can cause trouble with emails and potentially a spam relay point)

**Comment by illori** [comment]
how about something like sharethis widget and let that handle all the social network stuff in replace of the send topic to friends?

i dont know much about how it could work the dev side though http://www.sharethis.com

**Comment by Arantor** [comment]
I am not a fan of adding third party services to the core. I'd rather just ditch send topics and let the choice of mods be an option before going down that particular avenue. We can easily enough add simple sharing without having to use a third party system to do so.

**Comment by Arantor** [comment]
OK, so I looked again at the "simple view" and frankly it sucks.

When dealing with human/computer interaction, there are all kinds of things that need to be assessed, and one of those is the concept of information chunking: the number of options presented at once without any break. George A. Miller studied this back in the 1950s (especially discussing this stuff in The Psychology of Communication, 1967) and some of the things he found haven't actually changed since then, most notably the notion of dealing with approximately 7 items of data at once, ± 2, so 9 items in a single group at once then.

If you look at Classic Permissions, that number is largely consistent (because a single permission with own/any is still a single item) and the only major violation to this is the _Topics_ permissions list with 16 options, but by breaking that up into topics and topic moderation, we'd effortlessly solve that. (There are other minor violations, like _General_ and _Notifications_ with 2 and _Attachments_ with 3. But these can't really be merged anywhere anyway.)

On the other hand, Simple Permissions doesn't work out so well. Some areas are OK - _use basic functionality_, _moderate the entire forum_, these hit squarely in 7±2, but then we have _Personalize their profile_, with 3 items, and bizarrely, _Delete their account_ with just that one to itself..

But then we have _Moderate the board_ with 21 items. 21?! What the heck were people thinking with this? And because of this 'one line per permission' thing, it's actively hard to find things in it if you want to change something.

The rest of the board permissions aren't well grouped either, we have a couple of groups of 4 (_Post topics and replies to the board only after they have been approved_ and _View additional board content_ the latter of which doesn't make sense, why is "report posts to the moderators" a 'view additional board content' item? Why aren't the view/vote polls options with the post polls/add polls under a single polls item?) and why is _Post attachments_ a group of its own?

Seriously, there is nothing good about Simple Permissions. It violates the human-computer interface guidelines, it has things grouped together illogically, it raises the technical requirements for modders unnecessarily...

Quite honestly, I'd rather ditch the simple page and look at the other stuff for clean-up afterwards.

Unless anyone has any serious objections I WILL be removing it.

**Comment by illori** [comment]
+1 to removing it, i think it will make things easier all around. i wonder how many people actually use it or know it exists.

**Comment by emanuele45** [comment]
I think many that installed directly 2.0 know only the simple view that is the default. ;)
Not that I'm saying I like it (the first thing I usually do is switch) or I'm against removing it (I already removed it from Elk :angel: ).

**Comment by illori** [comment]
@emanuele45 you mean you are not against removing it? you left out a word ;)

**Comment by emanuele45** [comment]
Possible, I've just finished lunch, so... :innocent:

**Comment by Arantor** [comment]
Hmm, need to be careful with how to do this with the changes of #818 in it. The new profile permissions add a lot, breaking the 7±2 rule (there's now 14) and I'm trying to figure out how best to break that down without losing functionality.

**Comment by Arantor** [comment]
So simple permissions are now gone, I've rearranged the profile stuff a little but I'm still not entirely happy with it. Please try and see what you make of it.

**Comment by illori** [comment]
i think that will work. like most changes will take some time to get used to but i think worth it to make it so much easier to manage the permissions.

**Comment by AaronVanGeffen** [comment]
Glad to see it gone. Good riddance!

**Comment by Arantor** [comment]
The main issue is done, but the profile permissions do need some work to make them a bit more usable. But we'll think about it and open a new issue if we need to.
